### PR TITLE
[Merged by Bors] - TO-3110 [2] pass dummy dart migration data to rust

### PR DESCRIPTION
**Pull request by rustonaut**
Create a skeleton to passing dart migration data to rust, using them when initializing the db and cleaning up the dart db(s) afterward.

**Based On:**

- #577

### CODE REVIEW

**Comment by rustonaut** [comment]
@biernacki I added an additional commit which extracts the dart migration code into it's own module, sorry for that late change.

(It also now has the `DartMigrationData` class no longer under ffi, which stylistically wasn't that grate/uniform with the rest of the code.)

File: discovery_engine/lib/src/ffi/types/init_config.dart [hunk 1]
```diff
@@ -163,7 +163,7 @@ class InitConfigFfi with EquatableMixin {
     dataDir.writeNative(
       ffi.init_config_place_of_data_dir(place),
     );
-    useInMemoryDb.writeNative(ffi.init_config_place_of_use_in_memory_db(place));
+    useInMemoryDb.writeNative(ffi.init_config_place_of_use_ephemeral_db(place));
```
Comment (biernacki) [inline discovery_engine/lib/src/ffi/types/init_config.dart:166]: I think this needs a rebase :)

File: discovery_engine/lib/src/infrastructure/migration.dart [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        sourceReactedRepository.box.isEmpty) {
+      return null;
+    }
+
+    return DartMigrationData(
+      engineState: await engineStateRepository.load(),
+      documents: await documentRepository.fetchAll(),
+      activeDocumentData: activeDocumentDataRepository.box.toMap().map(
+            // ignore: avoid_annotating_with_dynamic
+            (dynamic key, data) =>
```
Comment (biernacki) [inline discovery_engine/lib/src/infrastructure/migration.dart:79]: Maybe you can try without dynamic
```suggestion
            (Object key, data) =>
```

**Review by biernacki — COMMENTED** [review]
Looks good, just needs a rebase :)

File: discovery_engine/lib/src/infrastructure/migration.dart (hunk 2 above)
Comment (rustonaut) [reply discovery_engine/lib/src/infrastructure/migration.dart:79]: sadly that doesn't work :sweat:

Comment (biernacki) [reply discovery_engine/lib/src/infrastructure/migration.dart:79]: maybe try `Object?` but if that doesn't work either, then just leave it as is

**Review by biernacki — APPROVED** [review]
lgtm

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**